include/RAJA/policy/cuda/reduce.hpp

```diff
@@ -70,21 +72,99 @@
 
 #include <cuda.h>
 
+#if defined(RAJA_ENABLE_OPENMP)
+#include <omp.h>
+#endif
+
 namespace RAJA
 {
 
-// HIDDEN namespace to encapsulate helper functions
-namespace HIDDEN
+namespace reduce
+{
+
+template <typename T>
+struct atomic_operators
+{
+
+};
+
+template <typename T>
+struct atomic_operators<sum<T>>
+{
+  RAJA_HOST_DEVICE RAJA_INLINE
+  void operator()(T &val, const T v)
+  {
+#ifdef __CUDA_ARCH__
+    RAJA::_atomicAdd(&val, v);
+#else
+    val += v;
```

Comment:
I meant in the code highlighted above. There's `val += v`. We already have a sum operation defined. e.g. `RAJA::reduce::sum<T>{}(val, v)`